src/pages/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {useEffect, useState} from "react";
+import axios from "axios";
+
+import styles from './MovieListPage.module.css';
+
+import SearchForm from "../../components/SearchForm/SearchForm";
+import GenreSort from "../../components/GenreSort/GenreSort";
+import MovieDetails from "../../components/MovieDetails/MovieDetails";
+import SortControl from "../../components/SortControl/SortControl";
+import Dialog from "../../components/Dialog/Dialog";
+import ModalContent from "../../components/ModalContent/ModalContent";
+
+import {GenreNames} from "../../constants/GenreNames";
+import {MovieDetailsData} from "../../types/MovieDetailsData";
+import AddMovieButton from "../../components/AddMovieButton/AddMovieButton";
+import MovieList from "../../components/MovieList/MovieList";
+
+export default function MovieListPage() {
+    const [movieList, setMovieList] = useState<MovieDetailsData[]>([]);
+    const [selectedMovie, setSelectedMovie] = useState<MovieDetailsData | null>(null);
+    const [showDialog, setShowDialog] = useState(false);
+    const [currentDialog, setCurrentDialog] = useState("");
+    const [isLoading, setIsLoading] = useState(false);
+    const [filters, setFilters] = useState({
+        searchQuery: "",
+        selectedGenre: "",
+        selectedSortControl: "release_date",
+    });
+
+    async function fetchMovies(filters: any, signal: AbortSignal) {
```

Comment:
mode this fetch function to separate file. Google where would be better to keep this function(services? )